### PR TITLE
ui: comment db updates

### PR DESCRIPTION
**Pull request by meowcodes**
meeting requirements below

Support comment and comment threads (1 level)
Each comment can be edited after posted
Each comment can be attached to a datadoc cell, or a whole table (make sure this system is extensible if we want to add commenting elsewhere)
When a cell is cloned or copied, the comment is not copied. When a cell is cut, the comment is moved
Support reactions, like Slack
Support richtext in comments

### CODE REVIEW

File: querybook/server/models/comment.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        sql.ForeignKey("comment.id", ondelete="CASCADE"),
+        nullable=False,
+        unique=True,
+    )
+
+    data_cell = relationship("DataCell", uselist=False)
+    comment = relationship("Comment", uselist=False)
+
+
+class NestedComment(CRUDMixin, Base):
```
Comment (jczhong84) [inline querybook/server/models/comment.py:89]: It seems this tables is not necessary to me. we can add this `parent_comment_id` to the `comment` table directly and `child_comment_id` is actually not needed.

**Comment by jczhong84** [comment]
also, do we need to introduce the concept of `resolve a comment`? and can a datacell have multiple comments?

File: querybook/server/models/comment.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    id = sql.Column(sql.Integer, primary_key=True, autoincrement=True)
+    comment_id = sql.Column(
+        sql.Integer,
+        sql.ForeignKey("comment.id", ondelete="CASCADE"),
+        index=True,
+        nullable=False,
+        unique=True,
+    )
+    reaction = sql.Column(
+        sql.String(length=name_length), index=True, unique=True, nullable=False
```
[the remark this answers is not in the document]
Comment (meowcodes) [reply querybook/server/models/comment.py:38]: it needs to be indexed so i can get the count easily

[the remark this answers is not in the document]
Comment (jczhong84) [reply querybook/server/models/comment.py]: as the count is always for a specific comment and `comment_id` is already indexed, and also there wont be too many reactions for a comment, so I dont think it's necessary to index it.

File: querybook/server/models/comment.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    created_by = sql.Column(
+        sql.Integer, sql.ForeignKey("user.id", ondelete="SET NULL"), nullable=True
+    )
+
+    text = sql.Column(sql.Text(length=mediumtext_length))
+
+    parent_comment_id = sql.Column(
+        sql.Integer,
+        sql.ForeignKey("comment.id", ondelete="CASCADE"),
+        index=True,
```
Comment (czgu) [inline querybook/server/models/comment.py:26]: why are we indexing on foreign keys? https://stackoverflow.com/questions/304317/does-mysql-index-foreign-key-columns-automatically

File: querybook/server/models/comment.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jczhong84) [reply querybook/server/models/comment.py]: to clarify, what meant is keep `index=True` for `comment_id`, while remove `index=True` for `reaction`

File: querybook/server/models/comment.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+class CommentReaction(CRUDMixin, Base):
+    __tablename__ = "comment_reaction"
+
+    id = sql.Column(sql.Integer, primary_key=True, autoincrement=True)
+    comment_id = sql.Column(
+        sql.Integer,
+        sql.ForeignKey("comment.id", ondelete="CASCADE"),
+        nullable=False,
+    )
+    reaction = sql.Column(sql.String(length=name_length), index=True, nullable=False)
```
Comment (czgu) [inline querybook/server/models/comment.py:39]: unless we are filtering on reaction, there is no point for an index? we would always filter it by comment id right?

File: querybook/server/models/comment.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        sql.ForeignKey("data_table.id", ondelete="CASCADE"),
+        nullable=False,
+    )
+    comment_id = sql.Column(
+        sql.Integer,
+        sql.ForeignKey("comment.id", ondelete="CASCADE"),
+        nullable=False,
+    )
+
+    data_table = relationship("DataTable", uselist=False)
```
Comment (czgu) [inline querybook/server/models/comment.py:60]: nit: remove these 2 relationships (i dont think this table will be used directly anyways)

File: querybook/server/models/comment.py (hunk 4 above)
Comment (meowcodes) [reply querybook/server/models/comment.py:39]: i guess we can aggregate reaction count on the front end

**Review by czgu — APPROVED** [review]

**Pull request merged**